### PR TITLE
botany balance changes

### PR DESCRIPTION
**Pull request by RafRoq**
this remove the extraction cap from yield and production speed based on stock parts

### CODE REVIEW

**Comment by yoyobatty** [comment]
right branch but why didn't you do the edits I told you to do?

**Comment by RafRoq** [comment]
> right branch but why didn't you do the edits I told you to do?

wait what edits, sorry

File: hippiestation/code/modules/hydroponics/gene_modder.dm [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+		else
+			max_potency = initial(max_potency) + (M.rating+12) // 62,74,86,98 	 Clamps at 100
+
+		max_yield = initial(max_yield)
+
+	for(var/obj/item/stock_parts/scanning_module/SM in component_parts)
+		if(SM.rating > 3) //If you create t5 parts I'm a step ahead mwahahaha!
+			min_production = 1
+		else
+			min_production = 1
```
Comment (danhibiki) [inline hippiestation/code/modules/hydroponics/gene_modder.dm]: if statement that yields same result regardless of value, shouldn't bother with this for loop

File: hippiestation/code/modules/hydroponics/gene_modder.dm [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+	..()
+	cut_overlays()
+	if((stat & (BROKEN|NOPOWER)))
+		icon_state = "dnamod-off"
+	else
+		icon_state = "dnamod"
+	if(seed)
+		add_overlay("dnamod-dna")
+	if(panel_open)
+		add_overlay("dnamod-open")
```
Comment (danhibiki) [inline hippiestation/code/modules/hydroponics/gene_modder.dm]: this whole chain isn't necessary, predefined in gene_modder.dm

File: hippiestation/code/modules/hydroponics/gene_modder.dm [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+			to_chat(user, "<span class='notice'>Please complete current operation.</span>")
+			return
+		eject_disk()
+		if(!user.transferItemToLoc(I, src))
+			return
+		disk = I
+		to_chat(user, "<span class='notice'>You add [I] to the machine.</span>")
+		interact(user)
+	else
+		..()
```
Comment (danhibiki) [inline hippiestation/code/modules/hydroponics/gene_modder.dm]: Unnecessary and predefined

File: hippiestation/code/modules/hydroponics/gene_modder.dm [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+				dat += "</table>"
+			else
+				dat += "No trait-related genes detected in sample.<br>"
+			if(can_insert && istype(disk.gene, /datum/plant_gene/trait))
+				dat += "<a href='?src=[REF(src)];op=insert'>Insert: [disk.gene.get_name()]</a>"
+			dat += "</div>"
+	else
+		dat += "<br>No sample found.<br><span class='highlight'>Please, insert a plant sample to use this device.</span>"
+	popup.set_content(dat)
+	popup.open()
```
Comment (danhibiki) [inline hippiestation/code/modules/hydroponics/gene_modder.dm]: Unnecessary

File: hippiestation/code/modules/hydroponics/gene_modder.dm [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+
+
+			update_genes()
+			operation = ""
+			target = null
+	else if(href_list["abort"])
+		operation = ""
+		target = null
+
+	interact(usr)
```
Comment (danhibiki) [inline hippiestation/code/modules/hydroponics/gene_modder.dm]: Unnecessary

File: hippiestation/code/modules/hydroponics/gene_modder.dm [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+
+/obj/machinery/plantgenes/proc/insert_seed(obj/item/seeds/S)
+	if(!istype(S) || seed)
+		return
+	S.forceMove(src)
+	seed = S
+	update_genes()
+	update_icon()
```
Comment (danhibiki) [inline hippiestation/code/modules/hydroponics/gene_modder.dm]: unnecessary

**Review by danhibiki — COMMENTED** [review]
You got a lot of stuff you don't need to define, and mostly you should be defining your base values for what your modding limits are on gene_modder machinery and the check for parts to manipulate potency.

**Comment by RafRoq** [comment]
@danhibiki  i did not made this code, this is the actual /tg/ and hippie code is using, i just changed numbers

**Comment by yoyobatty** [comment]
Uh, so why did you repaste tg code, why couldn't you just override the desired proc and put the var edits, you literally copy pasted the old cold and then editted numbers, this entire file already looks at the values from the tg version of gene_modder.dm. How it works is you're overwriting the code you want, not replacing the original file.

**Comment by RafRoq** [comment]
Gene_modder.dm did not existed in hippiestation/code/modules/hydroponics, you said to put it here, I did with the changes I made, the code is the same Hippie is using right now I don't understand what is the problem here

**Comment by bardofrage** [comment]
+1

**Comment by earthmoonjune** [comment]
+1

**Comment by earthmoonjune** [comment]
+1

**Comment by RafRoq** [comment]
@yoyobatty  i think its aproved to be merged but i forgot to add  hippiestation.dme @k-k-karma

**Comment by DaydreamPuppeh** [comment]
+1

**Review by yoyobatty — DISMISSED** [review]
This looks fine

File: hippiestation/code/modules/hydroponics/gene_modder.dm [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	max_yield = 10
+	min_production = 1
+	max_endurance = 40 // IMPT: ALSO AFFECTS LIFESPAN
+
+/obj/machinery/plantgenes/RefreshParts() // Comments represent the max you can set per tier, respectively. seeds.dm [219] clamps these for us but we don't want to mislead the viewer.
+	for(var/obj/item/stock_parts/manipulator/M in component_parts)
+		if(M.rating > 3)
+			max_potency = 100
+		else
+			max_potency = initial(max_potency) + (M.rating+12) // 62,74,86,98 	 Clamps at 100
```
Comment (yoyobatty) [inline hippiestation/code/modules/hydroponics/gene_modder.dm]: this doesn't do what you think it does, you want M.rating+12 to be M.rating*12

File: hippiestation/code/modules/hydroponics/gene_modder.dm [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+		max_yield = initial(max_yield)
+
+	for(var/obj/item/stock_parts/scanning_module/SM in component_parts)
+		max_endurance = initial(max_endurance) + (SM.rating * 25) // 35,60,85,100	Clamps at 10min 100max
```
Comment (yoyobatty) [inline hippiestation/code/modules/hydroponics/gene_modder.dm:15]: I think your comments are a little misleading

**Review by yoyobatty — CHANGES_REQUESTED** [review]

**Comment by yoyobatty** [comment]
Please reply to those review comments.

File: hippiestation/code/modules/hydroponics/gene_modder.dm (hunk 8 above)
Comment (RafRoq) [reply hippiestation/code/modules/hydroponics/gene_modder.dm:15]: I don't know what do toou mean by this one sorry

File: hippiestation/code/modules/hydroponics/gene_modder.dm (hunk 7 above)
Comment (RafRoq) [reply hippiestation/code/modules/hydroponics/gene_modder.dm]: Fixed

**Review by yoyobatty — APPROVED** [review]

**Pull request merged**